README.md

```diff
@@ -1,64 +1,148 @@
+**Remember**: Never commit this file or share it publicly.
 
-## 1. Install
-```shell
-> npm install
-// or
-> yarn install
-```
-For Bundler Integration Tests also install Docker. 
 
-## 2. Configure
-Place a mnemonic in a `.secret` file in the root folder of the project.
+### 🛠️ Development Commands
 
-## 3. Run
-All the smart contracts are carefully tested.
-There are two kinds of tests:
-* Standard Hardhat environment tests which test the main contracts logic
-* Bundler Integration tests which feature custom bundler-enabled environment to test that Smart Account and Modules operate properly in the wild with all the ERC-4337 limitations such as banned opcodes and storage access rules.
+Below are the commands you can use for various tasks:
 
-For Bundler integration tests you also need active Docker Environment.
+### 🧪 Testing
 
+Run regular tests:
 ```shell
-# Regular Tests
 npx hardhat test
+```
 
-# Bundler Integration Tests
-# Install realpath
+For Bundler Integration Tests, first install `realpath`:
+```shell
 brew install coreutils
-# Run Bundler Integration Tests
+```
+
+Then, run the Bundler Integration Tests:
+```shell
 yarn bundler-test
+```
 
-# other
-npx hardhat accounts
+### 📦 Compilation & Deployment
+
+Compile contracts:
+```shell
 npx hardhat compile
+```
+
+Clean the environment:
+```shell
 npx hardhat clean
+```
+
+Start a local Ethereum node:
+```shell
 npx hardhat node
+```
+
+Deploy contracts:
+```shell
+npx hardhat run scripts/deploy.ts
+TS_NODE_FILES=true npx ts-node scripts/deploy.ts
+```
+
+### 📈 Analysis & Reporting
+
+Display available accounts:
+```shell
+npx hardhat accounts
+```
+
+Get help on Hardhat commands:
+```shell
 npx hardhat help
+```
+
+Test with gas report:
+```shell
 REPORT_GAS=true npx hardhat test
+```
+
+Generate code coverage report:
+```shell
 npx hardhat coverage
-npx hardhat run scripts/deploy.ts
-TS_NODE_FILES=true npx ts-node scripts/deploy.ts
+```
+
+### 🧹 Code Quality & Formatting
+
+Lint JavaScript and TypeScript files:
+```shell
 npx eslint '**/*.{js,ts}'
+```
+
+Automatically fix linting issues:
+```shell
 npx eslint '**/*.{js,ts}' --fix
+```
+
+Check formatting for JSON, Solidity, and Markdown files:
+```shell
 npx prettier '**/*.{json,sol,md}' --check
+```
+
+Automatically format files:
+```shell
 npx prettier '**/*.{json,sol,md}' --write
+```
+
+Lint Solidity contracts:
+```shell
 npx solhint 'contracts/**/*.sol'
+```
+
+Automatically fix issues in Solidity contracts:
+```shell
 npx solhint 'contracts/**/*.sol' --fix
 ```
+---
+
+This format separates the description from the command, making it clearer and more readable.
+
+## 🔍 Etherscan Verification
+
+To verify on Etherscan, deploy a contract to an Ethereum network supported by Etherscan, like Ropsten. Set up your `.env` file, deploy your contract, and then verify:
+
+```shell
+hardhat run --network ropsten scripts/deploy.ts
+npx hardhat verify --network ropsten DEPLOYED_CONTRACT_ADDRESS "Hello, Hardhat!"
```

Comment:
Even for sake of example, we shouldn't use deprecated networks like ropsten. We can use goerli or sepolia here as a reference